### PR TITLE
Remove unused dist files

### PR DESCRIPTION
**Pull request by huwshimi**
In this branch:
- removed the modules.js definition (the YUI modules are now require()ed - this was empty in the last release, it generated `var YUI_MODULES=function(){return{}}();`)
- updated a bunch of modules to be require()ed rather than being loaded using the combo loader and were being attached window globals
- update the manifest to only require the files that need to be passed to the browser

### CODE REVIEW

**Comment by huwshimi** [comment]
[Remove unused dist files](https://trello.com/c/CkNfxTrL/730-remove-unused-dist-files)

**Comment by huwshimi** [comment]
QA:
- run `make dist`
- create a controller and update the controller to the generated dist
- open the gui and do a few basic things to make sure all is working as it should

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8774/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8775/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8776/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8778/

File: jujugui/static/gui/src/app/init/test-endpoint-utils.js [hunk 1]
```diff
@@ -558,12 +558,7 @@ describe('Application config handlers', () => {
         id: 'wordpress',
         pending: true,
         charm: charmUrl});
-      assert.equal(conn.messages.length, 1);
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/init/test-endpoint-utils.js]: I believe these counts must have been wrong before due to race conditions with other tests.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8779/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8782/

**Comment by hatched** [comment]
Thanks @huwshimi 

Smaller dist now :)

:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**